src/java.net.http/share/classes/jdk/internal/net/http/ConnectionPool.java

```diff
@@ -124,17 +126,22 @@ public int hashCode() {
     }
 
     ConnectionPool(long clientId) {
-        this("ConnectionPool("+clientId+")");
+        this(clientId, TimeSource.source());
     }
 
-    /**
-     * There should be one of these per HttpClient.
-     */
-    private ConnectionPool(String tag) {
+    ConnectionPool(long clientId, TimeSource timeSource) {
```

Comment:
Hello Daniel, as far as I can see, no one calls this constructor outside of this very class and this seems to be always passed a constant `TimeSource.source()`. So maybe we don't have to have a `timeSource` field in this class and instead just use `TimeSource.now()` at call sites within this class (and `ExpiryList`) whenever we want `now()`?

Reply:
This constructor should in fact take an InstantSource, not a TimeSource. It was added in case we wanted to upgrade the whitebox ConnectionPoolTest to be able to create an instance of ConnectionPool with a custom InstantSource.